From 72087c95fde5ac020483c039b7e34d27ec9699f8 Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Fri, 5 Aug 2022 11:45:22 +0100
Subject: [PATCH] Move find openmp in cmakelists

---
 CMakeLists.txt | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c231c7a..ec25aae 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,8 +45,6 @@ if(DEFINED ENV{CONDA_PREFIX})
     include_directories($ENV{CONDA_PREFIX}/include)
     link_directories($ENV{CONDA_PREFIX}/lib)
     link_directories($ENV{CONDA_PREFIX}/lib/intel64)
-else()
-    find_package(OpenMP)
 endif()
 
 # Add libraries
@@ -148,6 +146,7 @@ if(DEFINED ENV{CONDA_PREFIX} AND (NOT APPLE OR CMAKE_COMPILER_IS_GNUCC OR ENV{SK
     target_link_libraries("${TARGET_NAME}" PRIVATE gomp z)
 else()
     target_link_libraries("${TARGET_NAME}" PRIVATE ZLIB::ZLIB)
+    find_package(OpenMP)
     if(OpenMP_CXX_FOUND)
         target_link_libraries("${TARGET_NAME}" PRIVATE OpenMP::OpenMP_CXX)
     endif()
-- 
2.30.1 (Apple Git-130)

