From a2abae2659356bbea226ab5cf05a45e31ac223dd Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Wed, 20 Dec 2023 12:19:54 +0000
Subject: [PATCH] Fix for HDF5 linking on conda OS X

---
 CMakeLists.txt | 13 +++++++++++--
 1 file changed, 11 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ee34f5d..a2b3a57 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -150,8 +150,17 @@ if(CMAKE_CUDA_COMPILER)
 endif()
 target_link_libraries("${TARGET_NAME}" PRIVATE pybind11::module Eigen3::Eigen
     ${BLAS_LIBRARIES} gfortran m dl)
-if(DEFINED ENV{CONDA_PREFIX} AND (NOT APPLE OR CMAKE_COMPILER_IS_GNUCC OR ENV{SKETCHLIB_INSTALL} EQUAL "conda"))
-    target_link_libraries("${TARGET_NAME}" PRIVATE hdf5_cpp hdf5 gomp z)
+
+# HDF5 -- conda needs override
+if(DEFINED ENV{CONDA_PREFIX})
+    target_link_libraries("${TARGET_NAME}" PRIVATE hdf5_cpp hdf5)
+else()
+    target_link_libraries("${TARGET_NAME}" PRIVATE ${HDF5_LIBRARIES})
+endif()
+
+# OpenMP and zlib -- conda needs override; clang on OS X 'works' otherwise override
+if(NOT APPLE OR CMAKE_COMPILER_IS_GNUCC OR ENV{SKETCHLIB_INSTALL} EQUAL "conda")
+    target_link_libraries("${TARGET_NAME}" PRIVATE gomp z)
 else()
     target_link_libraries("${TARGET_NAME}" PRIVATE ${HDF5_LIBRARIES} ZLIB::ZLIB)
     find_package(OpenMP)
-- 
2.39.3 (Apple Git-145)

