From 7fdafd9a5e3cd9f621f8f9685ff315d62a0cdc0e Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Mon, 20 Apr 2020 18:04:39 +0100
Subject: [PATCH 2/2] Fix CUDA LDLIBS

---
 src/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile b/src/Makefile
index deafa7a..0e0027f 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -25,7 +25,7 @@ endif
 
 CPPFLAGS+=-I$(LIBLOC)/include -I$(LIBLOC)/include/eigen3
 LDFLAGS+= -L$(LIBLOC)/lib
-CUDA_LDLIBS=-lcudadevrt -lcudart_static $(LALIBS)
+CUDA_LDLIBS=-lcudadevrt -lcudart_static $(LDLIBS)
 
 CUDA_LDFLAGS =-L$(LIBLOC)/lib -L${CUDA_HOME}/targets/x86_64-linux/lib/stubs -L${CUDA_HOME}/targets/x86_64-linux/lib
 CUDAFLAGS =-Xcompiler -fPIC -Xptxas -dlcm=ca --cudart static --relocatable-device-code=true -gencode arch=compute_35,code=compute_35 -gencode arch=compute_50,code=sm_50 -gencode arch=compute_75,code=sm_75
-- 
2.25.0

