From ad17eb284365e571e9682e70ebab4f16d71bae29 Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Thu, 2 Jul 2020 17:18:27 +0100
Subject: [PATCH] Make rng constexpr for clang

---
 src/rng.hpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/rng.hpp b/src/rng.hpp
index d2f8c94..8f143dc 100644
--- a/src/rng.hpp
+++ b/src/rng.hpp
@@ -12,8 +12,8 @@ class Xoshiro {
     public:
         // Definitions to be used as URNG in C++11
         typedef size_t result_type;
-        static size_t min() { return std::numeric_limits<uint64_t>::min(); }
-        static size_t max() { return std::numeric_limits<uint64_t>::max(); }
+        static constexpr size_t min() { return std::numeric_limits<uint64_t>::min(); }
+        static constexpr size_t max() { return std::numeric_limits<uint64_t>::max(); }
         uint64_t operator()(); // generate random number U(min, max)
 
         // Constructor
-- 
2.25.0

