From 0e6115826312d806bd1a54dd56f42722bc552d72 Mon Sep 17 00:00:00 2001
From: John Lees <lees.john6@gmail.com>
Date: Wed, 17 Sep 2025 11:52:09 +0100
Subject: [PATCH] Update CUDA compute versions in makefile

---
 src/Makefile | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/Makefile b/src/Makefile
index 00719ef..25c0dd2 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -30,12 +30,7 @@ LDFLAGS+= -L$(LIBLOC)/lib
 CUDA_LDLIBS=-lcudadevrt -lcudart_static $(LDLIBS)
 
 CUDA_LDFLAGS =-L$(LIBLOC)/lib -L${CUDA_HOME}/targets/x86_64-linux/lib/stubs -L${CUDA_HOME}/targets/x86_64-linux/lib
-CUDAFLAGS +=-std=c++17 -Xcompiler -fPIC --cudart static --relocatable-device-code=true --expt-relaxed-constexpr -gencode arch=compute_70,code=sm_70 -gencode arch=compute_75,code=sm_75
-ifdef GPU
-	CXXFLAGS += -DGPU_AVAILABLE
-	CUDAFLAGS += -gencode arch=compute_86,code=sm_86
-	CUDA_LDFLAGS += -L${CUDA_HOME}/lib64
-endif
+CUDAFLAGS +=-std=c++17 -Xcompiler -fPIC --cudart static --relocatable-device-code=true --expt-relaxed-constexpr -gencode arch=compute_75,code=sm_75 -gencode arch=compute_86,code=sm_86 -gencode arch=compute_90,code=sm_90 -gencode arch=compute_100,code=sm_100
 
 PYTHON_LIB = pp_sketchlib$(shell python3-config --extension-suffix)
 
@@ -109,6 +104,7 @@ web/web_sketch.o: version.h
 python: $(PYTHON_LIB)
 
 $(PYTHON_LIB): $(SKETCH_OBJS) $(GPU_SKETCH_OBJS) $(CUDA_OBJS) sketchlib_bindings.o
+	CUDA_LDFLAGS += -L${CUDA_HOME}/lib64
 	nvcc $(CUDAFLAGS) $(CUDA_LDFLAGS) -Wno-deprecated-gpu-targets -shared -dlink $^ -o device_link.o -Xnvlink $(CUDA_LDLIBS)
 	$(LINK.cpp) $(CUDA_LDFLAGS) $(LDFLAGS) -shared $^ device_link.o -o $(PYTHON_LIB) $(CUDA_LDLIBS)
 
-- 
2.39.3 (Apple Git-146)

